### PR TITLE
Rename jog_arm->moveit_jog_arm

### LINKED ISSUE #1612: rename jog_arm to moveit_jog_arm
See https://github.com/ros-planning/moveit/pull/1606#discussion_r312545808 .

If anyone is not happy with the request to rename it, we can discuss here.

### PR DESCRIPTION
**Pull request by AndyZe**
Closes #1612

### CODE REVIEW

File: moveit_experimental/moveit_jog_arm/CMakeLists.txt [hunk 1]
```diff
@@ -37,27 +37,27 @@ include_directories(
...
 )
 
-add_executable(${PROJECT_NAME}_server
-  src/jog_arm/collision_check_thread.cpp
-  src/jog_arm/jog_arm_server.cpp
-  src/jog_arm/jog_calcs.cpp
-  src/jog_arm/jog_ros_interface.cpp
-  src/jog_arm/low_pass_filter.cpp
+add_executable(jog_arm_server
+  src/moveit_jog_arm/collision_check_thread.cpp
```
Comment (rhaschke) [inline moveit_experimental/moveit_jog_arm/CMakeLists.txt]: I suggest to skip this extra folder layer:
```suggestion
  src/collision_check_thread.cpp
```

File: moveit_experimental/moveit_jog_arm/CMakeLists.txt [hunk 2]
```diff
@@ -37,27 +37,27 @@ include_directories(
   ${EIGEN3_INCLUDE_DIRS}
 )
 
-add_executable(${PROJECT_NAME}_server
-  src/jog_arm/collision_check_thread.cpp
-  src/jog_arm/jog_arm_server.cpp
-  src/jog_arm/jog_calcs.cpp
-  src/jog_arm/jog_ros_interface.cpp
-  src/jog_arm/low_pass_filter.cpp
+add_executable(jog_arm_server
```
Comment (rhaschke) [inline moveit_experimental/moveit_jog_arm/CMakeLists.txt]: What about renaming this to `jog_server`?

File: moveit_experimental/moveit_jog_arm/include/moveit_jog_arm/collision_check_thread.h [hunk 3]
```diff
@@ -39,17 +39,17 @@
 
 #pragma once
 
-#include <jog_arm/jog_arm_data.h>
-#include <jog_arm/low_pass_filter.h>
+#include <moveit_jog_arm/jog_arm_data.h>
```
Comment (rhaschke) [inline moveit_experimental/moveit_jog_arm/include/moveit_jog_arm/collision_check_thread.h]: You are not going to install the headers, do you?
```suggestion
#include "jog_arm_data.h"
```

File: moveit_experimental/moveit_jog_arm/include/moveit_jog_arm/collision_check_thread.h [hunk 4]
```diff
@@ -39,17 +39,17 @@
...
+#include <moveit_jog_arm/low_pass_filter.h>
```
Comment (rhaschke) [inline moveit_experimental/moveit_jog_arm/include/moveit_jog_arm/collision_check_thread.h]: Same here and for other includes below.

File: moveit_experimental/moveit_jog_arm/include/moveit_jog_arm/collision_check_thread.h (hunk 3 above)
Comment (AndyZe) [reply moveit_experimental/moveit_jog_arm/include/moveit_jog_arm/collision_check_thread.h]: I take it that all of these header files need this change, not just this one?

Comment (AndyZe) [reply moveit_experimental/moveit_jog_arm/include/moveit_jog_arm/collision_check_thread.h]: I thought it was a ROS convention to put header files in include/${PACKAGE_NAME} so they won't clash with other packages if header files ever do need to be installed. Will ignore this suggestion for now.

Comment (AndyZe) [reply moveit_experimental/moveit_jog_arm/include/moveit_jog_arm/collision_check_thread.h]: (I appreciate learning the difference between "" and <> though)

**Pull request merged**